subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/repositories_cheat_sheet.adoc

```diff
@@ -0,0 +1,425 @@
+To use HTTP(S) header authentication, specify credentials with link:{javadocPath}/org/gradle/api/credentials/HttpHeaderCredentials.html[HttpHeaderCredentials] and use the link:{javadocPath}/org/gradle/authentication/http/HttpHeaderAuthentication.html[HttpHeaderAuthentication] scheme:
+
+====
+include::sample[dir="snippets/artifacts/defineRepository/groovy",files="build.gradle[tags=header-authenticated-maven-repo]"]
+include::sample[dir="snippets/artifacts/defineRepository/kotlin",files="build.gradle.kts[tags=header-authenticated-maven-repo]"]
+====
+
+[[sec:s3-repositories]]
+== Declare an AWS S3 Repository
+
+You can use the following system properties to configure interactions with s3 repositories:
+
+`org.gradle.s3.endpoint`::
+Used to override the AWS S3 endpoint when using a non AWS, S3 API compatible, storage service.
+
+`org.gradle.s3.maxErrorRetry`::
+Specifies the maximum number of times to retry a request in the event that the S3 server responds with a HTTP 5xx status code. When not specified a default value of 3 is used.
+
+[[sub:s3_url_formats]]
+=== S3 URL formats
+
+S3 URL's are 'virtual-hosted-style' and must be in the following format
+
+----
+s3://<bucketName>[.<regionSpecificEndpoint>]/<s3Key>
+----
+
+e.g. `s3://myBucket.s3.eu-central-1.amazonaws.com/maven/release`
+
+* `myBucket` is the AWS S3 bucket name.
+* `s3.eu-central-1.amazonaws.com` is the _optional_ http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region[region specific endpoint].
+* `/maven/release` is the AWS S3 key (unique identifier for an object within a bucket)
+
+
+[[sub:s3_proxy_settings]]
+=== S3 proxy settings
+
+A proxy for S3 can be configured using the following system properties:
+
+* `https.proxyHost`
+* `https.proxyPort`
+* `https.proxyUser`
+* `https.proxyPassword`
+* `http.nonProxyHosts` (NOTE: this is not a typo.)
+
+If the `org.gradle.s3.endpoint` property has been specified with a HTTP (not HTTPS) URI the following system proxy settings can be used:
+
+* `http.proxyHost`
+* `http.proxyPort`
+* `http.proxyUser`
+* `http.proxyPassword`
+* `http.nonProxyHosts`
+
+[[sub:s3_v4_signatures]]
+=== AWS S3 V4 Signatures (AWS4-HMAC-SHA256)
+
+Some of the AWS S3 regions (eu-central-1 - Frankfurt) require signing all HTTP requests in accordance with AWS's http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html[signature version 4].
+
+Always specify S3 URL's containing the region specific endpoint when using buckets that require V4 signatures:
+
+----
+s3://somebucket.s3.eu-central-1.amazonaws.com/maven/release
+----
+
+When a region-specific endpoint is _not_ specified for buckets requiring V4 Signatures, Gradle uses the default AWS region (us-east-1) and the
+following warning appears on the console:
+
+----
+Attempting to re-send the request to .... with AWS V4 authentication.
+To avoid this warning in the future, use region-specific endpoint to
```

Comment:
Thought: Is this the actual console message? it is missing an article or a plural?
```suggestion
To avoid this warning in the future, use a region-specific endpoint to
```
```suggestion
To avoid this warning in the future, use region-specific endpoints to
```

Reply:
I couldn't find the actual message in our source code, so I left it as-is.